radiation/python/radiation_driver_gt.py

```diff
@@ -0,0 +1,1270 @@
+        get_layer_temp(tem2da,
+                       tem2db,
+                       plyr,
+                       plvl,
+                       tlyr,
+                       tlvl,
+                       Statein["tgrs"],
+                       qlyr,
+                       delp,
+                       tvly,
+                       tem1d,
+                       tsfa,
+                       Sfcprop["tsfc"],
+                       Statein["qgrs"],
+                       ivflip,
+                       self.prsmin,
+                       self.QME5,
+                       self.QME6,
+                       self.lextop,
+                       domain=shape_nlp1,
+                       origin=default_origin,
+                       )
+
+        #  - Check for daytime points for SW radiation.
+
+        nday = 0
+        for i in range(IM):
+            if Radtend["coszen"][i,0] >= 0.0001:
+                nday += 1
+                idxday[nday - 1] = i + 1
+
+        #  - Call module_radiation_aerosols::setaer(),to setup aerosols
+        # property profile for radiation.
+
+        faersw, faerlw, aerodp = self.aer.setaer(
+            plvl,
+            plyr,
+            prslk1,
+            tvly,
+            rhly,
+            Sfcprop["slmsk"],
+            tracer1,
+            Grid["xlon"],
+            Grid["xlat"],
+            IM,
+            LMK,
+            LMP,
+            Model["lsswr"],
+            Model["lslwr"],
+        )
+
+        #  - Obtain cloud information for radiation calculations
+        #    (clouds,cldsa,mtopa,mbota)
+        #     for  prognostic cloud:
+        #    - For Zhao/Moorthi's prognostic cloud scheme,
+        #      call module_radiation_clouds::progcld1()
+        #    - For Zhao/Moorthi's prognostic cloud+pdfcld,
+        #      call module_radiation_clouds::progcld3()
+        #      call module_radiation_clouds::progclduni() for unified cloud and ncld=2
+
+        #  --- ...  obtain cloud information for radiation calculations
+
+        if Model["ncnd"] == 1:  # Zhao_Carr_Sundqvist
+            cloud_comp_1(ccnd, tracer1, ntcw-1, domain=shape_nlp1, origin=default_origin)
+        elif Model["ncnd"] == 2:  # MG
+            cloud_comp_2(ccnd, tracer1, ntcw-1, ntiw-1, domain=shape_nlp1, origin=default_origin)
+        elif Model["ncnd"] == 4:  # MG2
+            cloud_comp_4(ccnd, tracer1, ntcw-1, ntiw-1, ntrw-1, ntsw-1, domain=shape_nlp1, origin=default_origin)
+        elif Model["ncnd"] == 5:  # GFDL MP, Thompson, MG3
+            cloud_comp_5(ccnd, tracer1, ntcw-1, ntiw-1, ntrw-1, ntsw-1, ntgl-1, domain=shape_nlp1, origin=default_origin)
+
+        if Model["imp_physics"] == 11:
+            if not Model["lgfdlmprad"]:
+
+                cloud_comp_5_v2(ccnd, tracer1, ntcw-1, ntrw-1, ntiw-1, ntsw-1, ntgl-1, \
+                                self.EPSQ, domain=shape_nlp1, origin=default_origin)
+                # rsun the  summation methods and order make the difference in calculation
```

Comment:
Are all options tested? or is it just GFDL microphysics that's actually tested? If so, I would just `assert  Model["ncnd"] != 5` to limit the use of others, but you can keep those stencils. Also, replace stencil names to `get_cloud_properties`